### PR TITLE
Feature/navigation update

### PR DESCRIPTION
**Pull request by ZvonimirMedak**
## Summary

I have update the current state of the UserDefaults tool to match the changes done in the preferences tab, and vice versa. Also, I have updated the macOS navigation so that the users can now pop the view in the NavigationSplitView that's used on macOS instead of being open for no reason.

## Changes
The navigation changes are mostly done only for macOS since we needed to provide the selection binding from the SentinelListView.
The preference tab changes only added an onAppear to update the value if needed.
The UserDefaults changes are updating the view onAppear as well.

### Type

- [X] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [X] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [X] This pull request introduces a **breaking change**.

### Description
Everything was mentioned in the changes section

https://github.com/user-attachments/assets/e0da7bdf-85f4-44ca-bfa6-d3a081c5b1e8

## Checklist

- [X] I have performed a self-review of my own code.
- [X] I have tested my changes, including edge cases.
- [X] I have added necessary tests for the changes introduced (if applicable).
- [X] I have updated the documentation to reflect my changes (if applicable).

## Additional notes
-

### CODE REVIEW

File: Sentinel/Classes/UserDefaults/UserDefaultsToolViewModel.swift [hunk 1]
```diff
@@ -11,26 +11,70 @@ final class UserDefaultsToolViewModel: ObservableObject {
...
+                                    value: String(describing: value),
+                                    title: key,
+                                    userDefaults: userDefaults,
+                                    didDeleteProperty: nil // onAppear will update the screen on iOS
+                                )
+                            )
+                        }
+                    )
+                )
+                #endif
```
Comment (nikolamajcen) [inline Sentinel/Classes/UserDefaults/UserDefaultsToolViewModel.swift:74]: I would prefer moving `didSelect` into separate method, or just defining `didDeleteProperty` as a standalone variable and init the variable based on `if-else`.

File: Sentinel/Classes/UserDefaults/UserDefaultsToolDetailView.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                    })
+                }))
+                .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
+
+            Button(action: {
+                viewModel.didPressDelete()
+                #if os(macOS)
+                selection = nil
+                #else
+                presentationMode.wrappedValue.dismiss() // on macOS dismisses the whole window which isn't desired
```
Comment (nikolamajcen) [inline Sentinel/Classes/UserDefaults/UserDefaultsToolDetailView.swift:37]: I've seen this used multiple times; can we instead make an extension that will handle this case by calling it as a one liner on a calling side and define inner logic in the extension?

File: Sentinel/Classes/UserDefaults/UserDefaultsToolDetailView.swift [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+//
+
+import SwiftUI
+
+struct UserDefaultsToolDetailView: View {
+
+    @Environment(\.presentationMode) var presentationMode: Binding<PresentationMode>
+    @ObservedObject var viewModel: UserDefaultsToolDetailViewModel
+    #if os(macOS)
+    @Binding var selection: String?
```
Comment (nikolamajcen) [inline Sentinel/Classes/UserDefaults/UserDefaultsToolDetailView.swift:15]: Can we move this to the `viewModel`?

File: Sentinel/Classes/Core/PreferencesInfo/OptionToggleView.swift [hunk 4]
```diff
@@ -9,9 +9,10 @@ import SwiftUI
 
 struct OptionToggleView: View {
 
-    @State var value: Bool
+    @State private var value: Bool
     let title: String
     let onValueChanged: (Bool) -> Void
+    let getter: () -> Bool
```
Comment (nikolamajcen) [inline Sentinel/Classes/Core/PreferencesInfo/OptionToggleView.swift:15]: Which issue is fixed with adding a `getter` which is called on `onAppear`? Does `item.loadStoredValue()` not provide the correct value, and if not, in which cases?

**Review by nikolamajcen — CHANGES_REQUESTED** [review]
Good work! Still, check the comments I've added.

File: Sentinel/Classes/UserDefaults/UserDefaultsToolDetailView.swift (hunk 3 above)
Comment (ZvonimirMedak) [reply Sentinel/Classes/UserDefaults/UserDefaultsToolDetailView.swift:15]: Since it's a view specific binding, I feel like we should keep it here

File: Sentinel/Classes/UserDefaults/UserDefaultsToolViewModel.swift (hunk 1 above)
Comment (ZvonimirMedak) [reply Sentinel/Classes/UserDefaults/UserDefaultsToolViewModel.swift:74]: The difference between didSelect closures is messing up this one. The macOS one has a value provided in it which is then sent as the selection parameter, and the iOS one doesn't. Nothing comes to mind on how to handle this elegantly. Maybe you might think of something

File: Sentinel/Classes/UserDefaults/UserDefaultsToolDetailView.swift (hunk 2 above)
Comment (ZvonimirMedak) [reply Sentinel/Classes/UserDefaults/UserDefaultsToolDetailView.swift:37]: To extract this into a protocol or something we would still have to have if/else since the selection is hidden behind an if macos, and we would have to provide it somehow. It's called twice in the codebase, so for now, maybe we can leave it as is since macOS might be upped to version 13 so we can use NavigationStack

**Review by nikolamajcen — APPROVED** [review]
